common/types/hash.go

```diff
@@ -0,0 +1,35 @@
+package types
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	ethcommon "github.com/ethereum/go-ethereum/common"
+)
+
+// Hash represents a 32-byte hash
+type Hash = ethcommon.Hash
+
+// EmptyHash is an empty Hash
+var EmptyHash = Hash{}
+
+// IsHashZero returns true if the hash is all zeros, otherwise false
+func IsHashZero(h Hash) bool {
+	return h == EmptyHash
+}
+
+// HexToHash decodes a hex-encoded string into a hash
+func HexToHash(s string) (Hash, error) {
+	h, err := hex.DecodeString(strings.TrimPrefix(s, "0x"))
+	if err != nil {
+		return Hash{}, err
+	}
+	if len(h) != len(Hash{}) {
+		return Hash{}, fmt.Errorf("invalid len=%d hash", len(h))
+	}
+
+	var hash Hash
+	copy(hash[:], h)
+	return hash, nil
+}
```

Comment:
There is a `ethcommon.HexToHash` function that we could use instead, but it doesn't return an error value. The JSON marshaling for `ethcommon.Hash` does throw an error if the string is not hex or if the hex length is not 64 nibbles.